### PR TITLE
Manager partial update

### PR DESCRIPTION
**Pull request by Art3miX**
Allow updating service configs and authorizations for a workflow.

It accepts what updates the user wants to do, and output a list of `WasmMsg::ExecuteMsg` that needs to be executed by the owner in order to fulfil those changes.

The full process can be viewed in the `manager_test`.

### CODE REVIEW

**Review by keyleu — APPROVED** [review]
LGTM.

One question, we're not allowing to add new accounts with the update I guess right? In that case, if someone updates the workflow with new input/output accounts they will not be part of the registry and he needs to maintain that?

Also, want me to wait for this to be merged before I merge rename PR?

**Comment by Art3miX** [comment]
Yea, this partial update doesn't allow of modifying the program diagram (adding accounts or services), it is there in case you need some authorizations added/modified/disabled/enabled, and fix library configs that might be outdated or were set wrong. 
If you need more then that, you need to create a new program.

Regarding merging, whatever conflict will be easier to handle, I don't think it matters much, so whatever is easier to fix.

File: workflow-manager/src/helpers.rs
Comment (stiiifff) [inline workflow-manager/src/helpers.rs:39]: cleanup ?

File: workflow-manager/src/workflow_update.rs
Comment (stiiifff) [inline workflow-manager/src/workflow_update.rs:69]: A bit of a weird error .. Maybe you could create a constructor where you enforce that it's not possible to create WorkflowConfigUpdate with an `id` equal to zero (and any other basic validation).

File: workflow-manager/src/workflow_update.rs
Comment (stiiifff) [inline workflow-manager/src/workflow_update.rs:77]: This could be modelled as a `ChangeOwner` command / instruction.

File: workflow-manager/src/workflow_update.rs
Comment (stiiifff) [inline workflow-manager/src/workflow_update.rs:134]: Should you check if it doesn't exist yet before adding it ?

File: workflow-manager/src/workflow_update.rs
Comment (stiiifff) [inline workflow-manager/src/workflow_update.rs:257]: General comment: there's a lot happening in this function, and those are potentially sensitive operations (as in, breaking) on all aspects of a workflow/program config. I think it would be good (for future maintenance's sake) to organize the logic in a clear, staged structure .. as the current logic is mixing updating the authorizations, creating instructions, .. it's easy to get lost in what's happening.

File: workflow-manager/src/workflow_update.rs
Comment (Art3miX) [reply workflow-manager/src/workflow_update.rs:77]: Not sure what you mean.

File: workflow-manager/src/workflow_update.rs
Comment (Art3miX) [reply workflow-manager/src/workflow_update.rs:134]: Good point

File: workflow-manager/src/workflow_update.rs
Comment (Art3miX) [reply workflow-manager/src/workflow_update.rs:257]: It might not be clear what update does.

Update should not break the program, what it does is it allows updating library configs and authorizations, thats it, the manager cannot execute anything, and this is what instructions are, the update function output the messages that needs to be executed.

so there are 3 things happening here:
1. We build the messages to add/modify/disable/enable **authorizations**
2. We create the message to update the **library config**
3. Only the processor can update library configs because its the owner of the library contract, so we need to build the authorization that allows this operation.

After the update method ran, it output what messages needs to be executed by the owner in order to perform all those changes.

File: workflow-manager/src/workflow_update.rs
Comment (stiiifff) [reply workflow-manager/src/workflow_update.rs]: I mean that you are deriving whether to create a Change owner instruction from an optional field. The intent of modifying the owner could be modelled directly as a `ChangeOwner` instruction / command.

File: workflow-manager/src/workflow_update.rs
Comment (stiiifff) [reply workflow-manager/src/workflow_update.rs]: Maybe something more explicit like `InvalidWorkflowId` ?

File: workflow-manager/src/workflow_update.rs
Comment (stiiifff) [reply workflow-manager/src/workflow_update.rs]: I understand .. but as you mentioned, there are 3 things happening .. and the logic for those 3 things is interweaved / inter-related. I just suggest that making that code a bit clearer would help with future maintenance.

Comment (Art3miX) [reply workflow-manager/src/workflow_update.rs]: Added comments to explain a little more whats happeneing

File: program-manager/src/program_update.rs
Comment (stiiifff) [inline program-manager/src/program_update.rs:283]: This feels redundant ..

**Review by stiiifff — APPROVED** [review]

File: program-manager/src/program_update.rs
Comment (Art3miX) [reply program-manager/src/program_update.rs:283]: Yea, but explicit in what we verify, and we want to error here, so I felt like it will be better encapsulated in a helper function instead of bool and errors in code it self.
Should I change it?

**Pull request merged**
